### PR TITLE
Add black to config, CI

### PR DESCRIPTION
**Pull request by BSchilperoort**
This PR adds the "black" formatter to the project's configuration. In a separate commit I have formatted the code with black.

To not mess with our `git blame` history, I have added an ignore-revs file.

### CODE REVIEW

**Comment by bdestombe** [comment]
Two questions:
- Is black formatting guaranteed to pass the ruff tests?
- What line-length are we choosing?

**Comment by BSchilperoort** [comment]
> * Is black formatting guaranteed to pass the ruff tests?

Yes. There should never be a conflict between the two.[ "Ruff is compatible with Black out-of-the-box, as long as the line-length setting is consistent between the two."](https://beta.ruff.rs/docs/faq/)

>     * What line-length are we choosing?

Black's default is 88. The reasoning is [laid out here](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#line-length). In my experience that works quite well, and you can have two files open adjacently on a 1080p monitor.

**Comment by bdestombe** [comment]
Don't we also need to adjust the linting workflow/action?

**Comment by BSchilperoort** [comment]
> Don't we also need to adjust the linting workflow/action?

Nope. The linting GH Action uses the `hatch run lint` command, which is what I modified in pyproject.toml.

File: pyproject.toml [hunk 1]
```diff
@@ -182,6 +184,11 @@ skip = [".gitignore", ".tox", "docs"]
 src_paths = ["src", "tests"]
 line_length = 120
 
+[tool.black]
+line-length = 88
+target-version = ['py38', 'py39', 'py310']
```
Comment (bdestombe) [inline pyproject.toml:189]: 3.9 -> 3.11

**Review by bdestombe — CHANGES_REQUESTED** [review]

**Comment by bdestombe** [comment]
@BSchilperoort Lets also format the jupyter notebooks while we are at it.

It requires additional dependencies: "black[jupyter]"

**Comment by BSchilperoort** [comment]
> @BSchilperoort Let's also format the jupyter notebooks while we are at it.
> 
> It requires additional dependencies: "black[jupyter]"

Ok! In the end only 3 notebooks needed reformatting.

The tests are also formatted now. I have added an extra ignore-revs for those.

**Review by bdestombe — APPROVED** [review]
Looks good to me!

**Pull request merged**